### PR TITLE
Configure mac os to use proxy on connect and stop using on disconnect

### PR DESCRIPTION
**Pull request by furkhat**
Resolves privatix/privatix#137

### CODE REVIEW

File: plugin/adapter/client.go [hunk 1]
```diff
@@ -4,28 +4,59 @@ import (
...
+	"github.com/privatix/dapp-proxy/plugin/osconnector"
 	"github.com/privatix/dapp-proxy/plugin/v2ray-client"
 	"github.com/privatix/dappctrl/data"
 	"github.com/privatix/dappctrl/sess"
 )
 
+// host and port proxy is running on.
+const (
+	proxyHostname = "127.0.0.1"
+	proxyPort     = "1080"
```
Comment (sofabeat) [inline plugin/adapter/client.go]: Better to place in config

File: product/data/scripts/configuresocksfirewallproxy.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+if [ "$1" == "on" ]; then
+    if [ -f "$store_file" ]; then
+        echo "file '$store_file' exists" && exit 1
+    fi
+
+    host=$3
+    port=$4
+
+    networksetup -listallnetworkservices|grep -v "(*)"| while read in
```
[the remark this answers is not in the document]
Comment (sofabeat) [reply product/data/scripts/configuresocksfirewallproxy.sh]: We can use same technique of identifying internet looking interface by looking on default gateway. I think it exists in openvpn script

[the remark this answers is not in the document]
Comment (furkhat) [reply product/data/scripts/configuresocksfirewallproxy.sh]: I don't think I can configure proxy using `networksetup` based on network interfaces. 

HereIt is not about network interfaces, it is about network services. On my mac I have following network services
```
An asterisk (*) denotes that a network service is disabled.
Wi-Fi
Bluetooth PAN
Thunderbolt Bridge
```

For some machines that use cable to connect to internet there is also `Ethernet`.

[the remark this answers is not in the document]
Comment (sofabeat) [reply product/data/scripts/configuresocksfirewallproxy.sh]: It is possible to identify “connected” network services combining ‘ifconfig’ and ‘networksetup’ command. But to handle case, when user switching between Wi-Fi and cable connection it makes sense to leave it as you did.

**Pull request merged**